app/controllers/application_controller/filter.rb

```diff
@@ -947,7 +947,11 @@ def adv_search_build(model)
     # Restore @edit hash if it's saved in @settings
     @expkey = :expression                                               # Reset to use default expression key
     if session[:adv_search] && session[:adv_search][model.to_s]
-      @edit = copy_hash(session[:edit])
```

Comment:
@AparnaKarve As I am exploring your change, I realized that, yes, some condition is needed there but it's not as easy as it looks. We have to figure out, when exactly we should use `session[:adv_search][model.to_s]` and when `session[:edit]` because sometimes both are available but session[:edit] makes advanced search work properly (for example the situation in my PR https://github.com/ManageIQ/manageiq-ui-classic/pull/776 ). I mean, not always `session[:adv_search][model.to_s]` should be used even if it is available (I mean, not `nil`).